src/stan/mcmc/base_mcmc.hpp

```diff
@@ -1,38 +1,42 @@
 #ifndef STAN_MCMC_BASE_MCMC_HPP
 #define STAN_MCMC_BASE_MCMC_HPP
 
+#include <stan/interface_callbacks/writer/base_writer.hpp>
 #include <stan/mcmc/sample.hpp>
-
 #include <ostream>
 #include <string>
 #include <vector>
 
 namespace stan {
-
   namespace mcmc {
 
     class base_mcmc {
     public:
-      base_mcmc(std::ostream* o, std::ostream* e)
-        : out_stream_(o), err_stream_(e) {}
+      base_mcmc() {}
 
       virtual ~base_mcmc() {}
 
-      virtual sample transition(sample& init_sample) = 0;
+      virtual sample
+      transition(sample& init_sample,
+                 interface_callbacks::writer::base_writer& writer) = 0;
 
       std::string name() {
         return name_;
       }
 
-      virtual void write_sampler_param_names(std::ostream& o) {}
+      virtual void
+      write_sampler_param_names(interface_callbacks::writer::base_writer&
+                                writer) {}
 
-      virtual void write_sampler_params(std::ostream& o) {}
+      virtual void
+      write_sampler_params(interface_callbacks::writer::base_writer& writer) {}
 
```

Comment:
"write_sampler_param_names" and "write_sampler_params" should not be used anywhere and can be deleted.